include/ur_ddi.h

```diff
@@ -747,6 +784,185 @@ typedef ur_result_t(UR_APICALL *ur_pfnGetMemProcAddrTable_t)(
+typedef ur_result_t(UR_APICALL *ur_pfnBindlessImagesUnsampledImageCreateExp_t)(
+    ur_context_handle_t,
+    ur_exp_image_mem_handle_t,
+    const ur_image_format_t *,
+    const ur_image_desc_t *,
+    ur_mem_handle_t *,
+    ur_exp_image_handle_t *);
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Function-pointer for urBindlessImagesSampledImageCreateExp
+typedef ur_result_t(UR_APICALL *ur_pfnBindlessImagesSampledImageCreateExp_t)(
+    ur_context_handle_t,
+    ur_exp_image_mem_handle_t,
+    const ur_image_format_t *,
+    const ur_image_desc_t *,
+    ur_sampler_handle_t,
+    ur_mem_handle_t *,
+    ur_exp_image_handle_t *);
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Function-pointer for urBindlessImagesImageCopyExp
+typedef ur_result_t(UR_APICALL *ur_pfnBindlessImagesImageCopyExp_t)(
+    ur_context_handle_t,
+    void *,
+    void *,
+    const ur_image_format_t *,
+    const ur_image_desc_t *,
+    ur_exp_image_copy_flags_t,
+    uint32_t,
+    const ur_event_handle_t *,
+    ur_event_handle_t *);
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Function-pointer for urBindlessImagesImageGetInfoExp
+typedef ur_result_t(UR_APICALL *ur_pfnBindlessImagesImageGetInfoExp_t)(
+    ur_exp_image_mem_handle_t,
+    ur_image_info_t,
+    void *,
+    size_t *);
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Function-pointer for urBindlessImagesMipmapGetLevelExp
+typedef ur_result_t(UR_APICALL *ur_pfnBindlessImagesMipmapGetLevelExp_t)(
+    ur_context_handle_t,
+    ur_exp_image_mem_handle_t,
+    uint32_t,
+    ur_exp_image_mem_handle_t *);
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Function-pointer for urBindlessImagesMipmapFreeExp
+typedef ur_result_t(UR_APICALL *ur_pfnBindlessImagesMipmapFreeExp_t)(
+    ur_context_handle_t,
+    ur_exp_image_mem_handle_t);
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Function-pointer for urBindlessImagesImportOpaqueFDExp
+typedef ur_result_t(UR_APICALL *ur_pfnBindlessImagesImportOpaqueFDExp_t)(
+    ur_context_handle_t,
+    size_t,
+    uint32_t,
+    ur_exp_interop_mem_handle_t *);
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Function-pointer for urBindlessImagesMapExternalArrayExp
+typedef ur_result_t(UR_APICALL *ur_pfnBindlessImagesMapExternalArrayExp_t)(
+    ur_context_handle_t,
+    const ur_image_format_t *,
+    const ur_image_desc_t *,
+    ur_exp_interop_mem_handle_t,
+    ur_exp_image_handle_t *);
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Function-pointer for urBindlessImagesReleaseInteropExp
+typedef ur_result_t(UR_APICALL *ur_pfnBindlessImagesReleaseInteropExp_t)(
+    ur_context_handle_t,
```

Comment:
Will this interface work when we have multiple devices per `ur_context_handle_t`? Shouldn't there be a device argument instead?

Reply:
Basically when this is merged https://github.com/intel/llvm/pull/6446
you might have bugs if the interface is taking context when it should be taking device. Also more generally it could lead to bugs in backends other than cuda/hip.